cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/TransparentProxyDestination.java

```diff
@@ -0,0 +1,671 @@
+    }
+
+    @Nonnull
+    @Override
+    public Option<ProxyType> getProxyType()
+    {
+        return Option.none();
+    }
+
+    @Nonnull
+    @Override
+    public Option<KeyStore> getTrustStore()
+    {
+        return Option.none();
+    }
+
+    @Nonnull
+    @Override
+    public Option<String> getTrustStorePassword()
+    {
+        return Option.none();
+    }
+
+    /**
+     * Creates a new builder for a "static" destination.
+     * <p>
+     * A static destination connects directly to a specified URL and does not use the destination-gateway. It allows
+     * setting generic headers but does not support gateway-specific properties like destination name or fragments.
+     *
+     * @return A new {@link StaticBuilder} instance.
+     */
+    @Nonnull
+    public static StaticBuilder staticDestination( @Nonnull final String uri )
+    {
+        return new StaticBuilder(uri);
+    }
+
+    /**
+     * Creates a new builder for a "dynamic" destination that is resolved via the destination-gateway.
+     * <p>
+     * A dynamic destination requires a destination name and will be routed through the central destination-gateway. It
+     * supports all gateway-specific properties like fragments, tenant context, and authentication flows.
+     *
+     * @param destinationName
+     *            The name of the destination to be resolved by the gateway.
+     * @return A new {@link DynamicBuilder} instance.
+     */
+    @Nonnull
+    public static DynamicBuilder dynamicDestination( @Nonnull final String destinationName, @Nonnull final String uri )
+    {
+        return new DynamicBuilder(destinationName, uri);
+    }
+
+    /**
+     * Abstract base class for builders to share common functionality like adding headers and properties.
+     *
+     * @param <B>
+     *            The type of the builder subclass, used for fluent method chaining.
+     */
+    @Accessors( fluent = true, chain = true )
+    public abstract static class AbstractBuilder<B extends AbstractBuilder<B>>
+    {
+        final List<Header> headers = new ArrayList<>();
+        final List<DestinationHeaderProvider> customHeaderProviders = new ArrayList<>();
+        final DefaultDestination.Builder propertiesBuilder = DefaultDestination.builder();
+
+        /**
+         * Returns the current builder instance.
+         * <p>
+         * This method is used to support fluent method chaining in subclasses of {@code AbstractBuilder}.
+         *
+         * @return the current builder instance
+         */
+        protected abstract B getThis();
+
+        /**
+         * Adds the given key-value pair to the destination to be created. This will overwrite any property already
+         * assigned to the key.
+         *
+         * @param key
+         *            The key to assign a property for.
+         * @param value
+         *            The property value to be assigned.
+         * @return This builder.
+         * @since 5.0.0
```

Comment:
```suggestion
         * @since 5.22.0
```

and apply the same everywhere for consistency